lara-typescript/src/page-item-authoring/mw-interactives/customize.tsx

```diff
@@ -40,34 +59,73 @@ export const CustomizeMWInteractive: React.FC<Props> = (props) => {
     height: native_height,
     mode: (aspect_ratio_method || "DEFAULT") as AspectRatioMode
   });
-  const [clickToPlay, setClickToPlay] = useState(click_to_play);
-  const [enableLearnerState, setEnableLearnerState] = useState(enable_learner_state);
+
+  const [state, setState] = useState({
+    aspect_ratio_method,
+    click_to_play,
+    click_to_play_prompt,
+    enable_learner_state,
+    full_window,
+    image_url,
+    native_height,
+    native_width,
+  });
+
+  useEffect(() => {
+    setState({
+      ...state,
+      aspect_ratio_method: aspectRatioValues.mode,
+      native_height: aspectRatioValues.height,
+      native_width: aspectRatioValues.width
+    });
```

Comment:
```suggestion
    setState(state => ({
      ...state,
      aspect_ratio_method: aspectRatioValues.mode,
      native_height: aspectRatioValues.height,
      native_width: aspectRatioValues.width
    }));
```
Any time the new state depends on the previous state (like here and several places below) you should use the functional form of `setState()`, otherwise you run the risk of getting stale values.

Reply:
Thanks, @kswenson 👍 That probably explains why one aspect of the edit form wasn't quite working right. We've decided to approach handling the form data differently anyway, but that's definitely still good to know generally.